### ISSUE TITLE
Architecture questions

### ISSUE THREAD

**Issue by zhangdoa** [thedmd/imgui-node-editor#51]
Hello thedmd,

First of all, thanks for starting this amazing project. I've been investigating the node editor solutions for a while in order to integrate into one of my projects, and I found yours and tried a little bit. Since I used ImGui previously too, I'd like to ask some questions because the state of this project now is still not easy to integrate into other projects:

1. Would you like to separate the platform-specific impl of ImGui (rendering impl and window impl) from the Example folder? It should not be the user's obligation to fully care about how the platform-specific things done if your dependencies restricted the platform-specific impl to their design. And it's more flexible to isolate them into separate libs rather than coupling with the entry module, then it's easier to let the user use build config to choose platform and rendering backend; 

2. Would you like to use the relative path to include third-party headers in `imgui_node_editor.h`? As a user, I don't want to intrusively modified my build configs to fit your build configs, it's better to only have physical dependencies among headers rather than mess with makefiles;

3. What's the future direction of this project? Would it have a self-constraint API or would it keep mixing with ImGui?

Thanks and wish you a happy new year,

**Comment by thedmd** [comment]
Thanks.

1) Editor is in `NodeEditor` directory and will work with your copy of ImGui. It is examples who has dependencies. I stated that in readme, but that may be not enough and I should reorganize sources to more reflect ImGui structure.
I learned people often treat blueprint example (build on top of node editor) as a starting point. This one indeed relies on not merged PR. Is that true also for you?
Can you tell me how do you approach integration? I wish to remove obstacles I failed to see already.

2) Do you mean to have `# include "imgui.h"` instead of `# include <imgui.h>`?

3) Editor is build on top of ImGui. Intention is not to hide that fact but to serve as an extension. I'm thinking about it as a yet another widget and tried to make it behave as one.
Editor API grown organically with project development. What made perfect sense then may not be obvious now. I have in mind some improvements I can make. Good think is I keep API functions separated from implementation, so making improvements should not introduce breakages.

I wish you happy new year too.

**Comment by zhangdoa** [comment]
Thanks for the reply,

I thought previously this project is an individual module over ImGui, that I don't need to interact with the ImGui impls. But as you explained and now I understood, whether I need to have an ImGui submodule or use the version you provided, because the project I want to integrate into doesn't have anything related to ImGui yet.
I took a look at all the examples and my major questionary was all around the Examples\Common\Application because initially, I didn't understand why you didn't encapsulate the platform-specific stuff inside the NodeEditor, now it's not a problem.
Previously I added a wrapper module in my project, included the imgui_node_editor.h and tried to link against the NodeEditor libs, of course, it won't work without an ImGui impl lib. Now I would use this project as a vanilla ImGui extension.

Basically yes, because if I'd use my version of ImGui and the NodeEditor, then it would force me to modify my makefile to add ImGui include directories or modify the #include <imgui.h> to something like #include ../../imgui/imgui.h. But now I think it might be an inevitable modification.

Thanks for the clarification, I'd try to integrate to further explore the possibilities.
